apps/charterafrica/src/payload/collections/Tags.js

```diff
@@ -17,7 +17,30 @@ const Tags = {
       localized: true,
       required: true,
     },
-    slug(),
+    slug({ fieldToUse: "name" }),
+    {
+      name: "collectionType",
```

Comment:
Do we need this? Doesn't payload allow offer APIs to query tags from news collection separate from research? I mean there is already a relationship right?

My point being: a tag `Africa` should be reusable in news, research, grants, etc., without having to duplicate it.

Reply:
When creating a  slug, we may need to specify which collection(s) does this slug apply to.

This is necessary for filtering purposes.

Perhaps maybe have this field as  `hasMany`. @kilemensi

Reply:
I still don't understand the problem @koechkevin so I can't really say if `hasMany` will help or not.

Can you give a scenario where we need to specify the collection(s) the slug applies to?

Reply:
Yes but this is where it breaks down @koechkevin : if in News there is Kenya tag and now there is Grantee from Kenya and content team wants to label the Grantee using Kenya tag.

Isn't this approach going to create a new Tag altogether?

My thinking was tags can be shared in the CMS but on the front end, we'll have to filter tags by collection i.e. give me all tags that have been used in News. I'm assuming the relationship field would allow us to do that.

Isn't this the case?

Reply:
Got it.

But I think that would be a bit complex.

Reply:
That's why we're having this discussion @koechkevin ... Why do you think it will be complex?

Reply:
The structure of a collection does not include the collections in which it has been linked to.

To fetch on the frontend would therefore mean fetching the collection e.g news itself then pick all the tags uniquely which would be an expensive operation @kilemensi

Reply:
🤔 ... so if you do say `collection?slug=news&depth=1 (or 2)`, wouldn't it include `tags` as well?

But, my main issue is I don't want us to duplicate tags just per collection. The proof is in the pudding, so may be let me pull and play with it locally.

Reply:
Having gone through this, I think lets stick to the old design i.e. no filter. Yes, it might be expensive to make extra call but I believe tags are just markers. We shouldn't allow duplication across collections.

May be we should even add a `unique: true` to the name field?